lib/spree/retailops/rop_order_helper.rb

```diff
@@ -141,7 +141,7 @@ def calculate_ship_price(removed = false)
         # This resolves https://github.com/dotandbo/dotandbo-spree/issues/3857
         # Where entire shipping fee is removed from order if 
         # last unshipped line_item is canceled.
-        if removed && !method && @order.shipments.present? && !@order.shipments.unshipped.present?
+        if removed && !method && @order.shipments.present? && @order.shipments.where('state != ?','shipped').empty?
```

Comment:
Yeah. I think it makes sense to add the scope to the model. We can then reuse the scope if it's needed elsewhere and just in case the definition of unshipped changes in the future, we only need to change it in one place.